dhis-2/dhis-support/dhis-support-external/src/main/java/org/hisp/dhis/external/conf/ConfigurationKey.java

```diff
@@ -149,7 +149,16 @@
     AUDIT_ENABLED( "system.audit.enabled", Constants.TRUE, false ),
     TRACKER_IMPORT_PREHEAT_CACHE_ENABLED( "tracker.import.preheat.cache.enabled", Constants.ON, false ),
     ENABLE_OAUTH2_AUTHORIZATION_SERVER( "oauth2.authorization.server.enabled", Constants.ON, false ),
-    ENABLE_JWT_OIDC_TOKEN_AUTHENTICATION( "oidc.jwt.token.authentication.enabled", Constants.OFF, false );
+    ENABLE_JWT_OIDC_TOKEN_AUTHENTICATION( "oidc.jwt.token.authentication.enabled", Constants.OFF, false ),
+    CSP_NONCE_ENABLED( "csp.nonce.enabled", Constants.ON, true ),
+    CSP_UPGRADE_INSECURE_ENABLED( "csp.upgrade.insecure.enabled", Constants.OFF, true ),
+    CSP_HEADER_VALUE( "csp.header.value", "prefetch-src 'self'; "
+        + "script-src 'self' 'nonce-{nonce}' 'unsafe-inline'; "
+        + "connect-src 'self' *.fastly.net; img-src 'self' data: *.fastly.net; "
```

Comment:
Does this affect things like OpenStreetMap, Bing maps, Google Earth Engine (used in Maps)?

Reply:
Yes, we need to add this either as default (static) or maybe make it optional aka. more dynamic. It's still a lot to figure out here. I'm trying to set up a CSP multi-end taskforce for this, so we can coordinate.